janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/ElasticSearchIndex.java

```diff
@@ -693,12 +724,23 @@ public void mutate(Map<String, Map<String, IndexMutation>> mutations, KeyInforma
                                 upsert = null;
                             }
 
-                            final String inline = getAdditionScript(information, storeName, mutation);
-                            if (!inline.isEmpty()) {
-                                final ImmutableMap.Builder builder = compat.prepareScript(inline);
-                                requestByStore.add(ElasticSearchMutation.createUpdateRequest(indexStoreName, storeName,
-                                        documentId, builder, upsert));
-                                log.trace("Adding script {}", inline);
+                            if (compat.supportsParams()) {
+                                List<Map<String, Object>> params = getParameters(information.get(storeName), mutation.getAdditions(),
+                                        c -> c == Cardinality.SET || c == Cardinality.LIST);
```

Comment:
I basically don't know anything about script parameterization with ES, so this might be a stupid question, but why does this only work for cardinality `SET` or `LIST` and how about indexed properties with `Cardinality.SINGLE`? How are they added?

Reply:
`SINGLE` will just replace the existent value and there is no need in scripts since we're not merging anything.